### ISSUE TITLE
Swagger - generate proper type information for maps

### ISSUE THREAD

**Issue by mwielgus** [kubernetes/kubernetes#13199]
The current spec results in creating a placeholder-type objecty "Any".

**Comment by adynathos** [comment]
I am trying to generate a Java client library for Kubernetes using the published swagger spec and [swagger-codegen](https://github.com/swagger-api/swagger-codegen).

The spec contains invalid type information ```"type": "any"``` for fields that should contain maps.

Example: 
* ```ResourceRequirements``` contains field ```limits``` which has type ```map[ResourceName]resource.Quantity``` (```ResourceName``` is a string).

* In the swagger spec file v1.json, the definition for this field has invalid type information:
```
"v1.ResourceRequirements": {
    ...
    "properties": {
     "limits": {
      "type": "any",
      "description": "..."
     },
     ...
    }
   }
```

* When a Java client library is generated from this spec, it contains reference to class ```Any``` which of course does not exist:
```
import io.kubernetes.client.model.Any;
...
public class V1ResourceRequirements   {  
  private Any limits = null;
```

**Proposed solution**
* Maps are supported in Swagger 2.0. To use them, the spec should be migrated to Swagger 2.0. This can be done using the [swagger-converter](https://github.com/apigee-127/swagger-converter) program.

* Specify proper type information for maps using ```additionalProperties```. For example, this way we could define a string-to-string map (for JSON, this will always have to be a string-to-something map):
```
"v1.ResourceRequirements": {
	...
	"properties": {
		"limits": {
			"type": "object",
			"additionalProperties": {
				"type": "string"
			}
		},
```
The generated Java class then has the proper type:
```
public class V1ResourceRequirements   {
  private Map<String, String> limits = new HashMap<String, String>();
```

**Comment by mwielgus** [comment]
cc: @nikhiljindal

**Comment by nikhiljindal** [comment]
Yes, this was discussed before in https://github.com/kubernetes/kubernetes/issues/4700.
This will be fixed when we move to swagger 2.0: https://github.com/kubernetes/kubernetes/issues/13414

Closing this as a dup of https://github.com/kubernetes/kubernetes/issues/4700
